Author: Alastair McKinstry <mckinstry@debian.org>
Description: Fix format-security error.
Last-Updated: 2011-10-23
Forwarded: no

Index: grads-2.2.1/src/gagx.c
===================================================================
--- grads-2.2.1.orig/src/gagx.c
+++ grads-2.2.1/src/gagx.c
@@ -4977,11 +4977,11 @@ gadouble *gr;
 
   if (pcm->ffile != stdout) {
     if (pcm->fwname) {
-      snprintf(pout,1255,"Wrote %ld of %i elements to ", written, exsz);
+      snprintf(pout,1255,"Wrote %ld of %i elements to ", (long) written, exsz);
       gaprnt (2,pout);
       gaprnt (2,pcm->fwname);
     } else {
-      snprintf(pout,1255,"Wrote %ld of %i elements to grads.fwrite", written, exsz);
+      snprintf(pout,1255,"Wrote %ld of %i elements to grads.fwrite", (long) written, exsz);
       gaprnt (2,pout);
     }
   
